src/telemetry.ts

```diff
@@ -0,0 +1,57 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+'use strict';
+
+import * as vscode from 'vscode';
+import * as util from './util';
+import TelemetryReporter from 'vscode-extension-telemetry';
+
+interface IPackageInfo {
+    name: string;
+    version: string;
+    aiKey: string;
+}
+
```

Comment:
consider adding:

```ts
export type Properties = { [key: string]: string };
export type Measures = { [key: string]: number };
```

and using them as the types for the function below.

Reply:
Sure that's a good idea.